pkg/ottl/contexts/internal/ctxprofilesample/profilesample.go

```diff
@@ -0,0 +1,115 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package ctxprofilesample // import "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl/contexts/internal/ctxprofilesample"
+
+import (
+	"context"
+
+	"go.opentelemetry.io/collector/pdata/pprofile"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl/contexts/internal/ctxerror"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl/contexts/internal/ctxutil"
+)
+
+type ProfileSampleContext interface {
+	GetProfileSample() pprofile.Sample
+}
+
+func PathGetSetter[K ProfileSampleContext](path ottl.Path[K]) (ottl.GetSetter[K], error) {
+	if path == nil {
+		return nil, ctxerror.New("nil", "nil", Name, DocRef)
+	}
+	switch path.Name() {
```

Comment:
"attributes" are missing, but listed in README.md (which makes sense to me).
It should be similar to what we have for `profile`: https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/07f58dcf39cba89e61b5c1f5ecfaaef6ed935f02/pkg/ottl/contexts/internal/ctxprofile/profile.go#L70